src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,27 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import java.util.NoSuchElementException;
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.dao.impl.CountryDaoImpl;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import org.hibernate.SessionFactory;
 
 public class CountryServiceImpl implements CountryService {
+    private CountryDao countryDao;
+
+    public CountryServiceImpl(SessionFactory sessionFactory) {
+        countryDao = new CountryDaoImpl(sessionFactory);
+    }
```

Comment:
```suggestion
    public CountryServiceImpl(CountryDao countryDao) {
        this.countryDao = countryDao;
    }
```

Reply:
But we have the next code with the comment:
```
public class Main {
    public static void main(String[] args) {
        // use this session factory when you will **initialize service instances**
        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
```
Service, not dao